rustls/src/client/hs.rs

```diff
@@ -1228,3 +1245,85 @@ fn low_quality_integer_hash(mut x: u32) -> u32 {
     x = (x ^ 0xb55a4f09) ^ (x >> 16);
     x
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::msgs::enums::{CertificateType, ExtensionType};
+    use crate::{Error, PeerIncompatible};
+
+    use super::check_certificate_type;
+
+    #[test]
+    fn test_check_certificate_type() {
```

Comment:
Not sure this test actually carries its weight? The compiler already checks that the `match` expression in `check_certificate_type()` is exhaustive, and I doubt whether adding an exact transliteration of the `match` in test form is actually useful.

Is this important to coverage? Can we generate the coverage in a more "real"/integrated way?

Reply:
I agree with you. The tests in `api.rs` should already test most arms in an integrated way. 
Oddly enough before adding these tests the code coverage was complaining, after removing these tests and a new push it passes...